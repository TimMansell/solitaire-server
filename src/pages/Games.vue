<template>
  <GameOverlay
    ref="historyOverlay"
    center-content
    show-close
    data-test="history-overlay"
  >
    <template #title> History </template>
    <template #msg>
      <GameHistory v-if="userGameCount" @scrollTo="scrollTo" />
      <p v-if="!userGameCount" data-test="game-history-no-games-msg">
        You have not played any games yet
      </p>
    </template>
  </GameOverlay>
</template>

<script>
import { mapGetters } from 'vuex';
import VueScrollTo from 'vue-scrollto';
import GameOverlay from '@/components/GameOverlay.vue';
import GameHistory from '@/components/GameHistory.vue';

export default {
  name: 'HistoryOverlay',
  components: {
    GameOverlay,
    GameHistory,
  },
  computed: {
    ...mapGetters(['userGameCount']),
  },
  methods: {
    scrollTo(elem) {
      const { historyOverlay } = this.$refs;

      VueScrollTo.scrollTo(elem, {
        container: historyOverlay.$el,
        offset: -10,
      });
    },
  },
};
</script>
