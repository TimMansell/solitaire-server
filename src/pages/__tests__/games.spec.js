import { shallowMount } from '@vue/test-utils';
import Games from '@/pages/Games.vue';

const mocks = {
  $store: { dispatch: jest.fn() },
};

describe('Games.vue', () => {
  it('matches snapshot - no games', () => {
    const wrapper = shallowMount(Games, {
      mocks,
      computed: {
        userGameCount: () => 0,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot - with games', () => {
    const wrapper = shallowMount(Games, {
      mocks,
      computed: {
        userGameCount: () => 4,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should not show no games message', () => {
    const wrapper = shallowMount(Games, {
      mocks,
      computed: {
        userGameCount: () => 4,
      },
    });

    expect(
      wrapper.find('[data-test="game-history-no-games-msg"]').exists()
    ).toBe(false);
  });

  it('should show no games message', () => {
    const wrapper = shallowMount(Games, {
      mocks,
      computed: {
        userGameCount: () => 0,
      },
    });

    expect(
      wrapper.find('[data-test="game-history-no-games-msg"]').exists()
    ).toBe(true);
  });
});
