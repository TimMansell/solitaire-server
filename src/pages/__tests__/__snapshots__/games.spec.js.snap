// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Games.vue matches snapshot - no games 1`] = `
<gameoverlay-stub centercontent="true" showclose="true" data-test="history-overlay"> History
  <!---->
  <nogames-stub data-test="game-history-no-games-msg"></nogames-stub>
</gameoverlay-stub>
`;

exports[`Games.vue matches snapshot - with games 1`] = `
<gameoverlay-stub centercontent="true" showclose="true" data-test="history-overlay"> History <gamehistory-stub></gamehistory-stub>
  <!---->
</gameoverlay-stub>
`;
